src/Tokenizer.ts

```diff
@@ -0,0 +1,192 @@
+        if ( this.raw[idx] !== '(') {
+            throw new MethodParamsTokenizerException('Expected ( at the beginning of parameter list declaration.');
+        }
+
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_LIST_BEGIN, this.raw[idx]));
+        idx = this._ignore(/\s/, idx + 1);
+        while (idx < this.raw.length && this.raw[idx] !== ')') {
+            if (this.tokens.length > 1) {
+                if (this.raw[idx] !== ',') {
+                    throw new MethodParamsTokenizerException('Expected , between parameter definitions.');
+                }
+                this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_SEPARATOR, ','));
+                idx = this._ignore(/\s/, idx + 1);
+            }
+            idx = this.parameter(idx);
+        }
+
+        if (idx >= this.raw.length)
+            throw new MethodParamsTokenizerException('Expected ) at the end of parameter list declaration.');
+
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_LIST_END, this.raw[idx]));
+
+        const excessStart = idx + 1;
+        const excess = this.raw.slice(excessStart);
+        if (excess.match(/[^\s]/g)) {
+            throw new MethodParamsTokenizerException(`Unexpected out of bound expression ${excess}.`);
+        }
+    }
+
+    parameter(idx:number):number {
+        idx = this.parameterName(idx);
+
+        if (this.raw[idx] !== ':') {
+            const lastName = this.tokens[this.tokens.length - 1].value;
+            throw new MethodParamsTokenizerException(`Expected : after parameter ${lastName}.`);
+        }
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_NAME_VALUE_SEPARATOR, this.raw[idx]));
+
+        idx = this._ignore(/\s/, idx + 1);
+        try {
+            idx = this.parameterValue(idx);
+        } catch (e) {
+            const paramName = this.tokens[this.tokens.length - 2].value;
+            e.message = `${e.message} in parameter ${paramName}.`;
+            throw e;
+        }
+        return this._ignore(/\s/, idx);
+    }
+
+    parameterName(idx:number):number {
+        const nameEnd = this._ignore(/\w/, idx);
+        const name = this.raw.slice(idx, nameEnd);
+        if (!name)
+            throw new MethodParamsTokenizerException(`Expected parameter name, found ${this.raw[idx]}`);
+
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_NAME, name));
+        return nameEnd;
+    }
+
+    parameterValue(idx:number):number {
+        if (this.raw[idx].match(/('|")/))
+            return this.stringLiteral(idx);
+
+        const valueEnd = this._until(/\s|,|\)/, idx);
+        const value = this.raw.slice(idx, valueEnd);
+        if (!this._checkPrimitiveValue(value))
+            throw new MethodParamsTokenizerException(`Invalid value ${value}`);
+
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_VALUE, value));
+        return valueEnd;
+    }
+
+    stringLiteral(idx:number):number {
```

Comment:
This should be valid? `@response (vary: "x-vtex-,segment", test: "\\"test")`

Reply:
Very nice!

Reply:
I've elaborated something better than iterating with `_until()` that I think solves this, in case you have time, please take a look and try breaking it! 😄